### PR TITLE
Creating document index from terraform

### PR DESCRIPTION
**Pull request by dvargasfuertes**
https://www.loom.com/share/fd8dc1e14e5449a1a4f785702a6f45eb

We can create a document index from terraform!! 🎉 

Next steps are going to be on the Vellum Side:
- Update API
- Delete API
- Make Indexing Config optional in Create

### CODE REVIEW

File: internal/provider/document_index/data.go [hunk 1]
```diff
@@ -2,13 +2,13 @@ package document_index
...
 	"context"
-	"fmt"
-	"net/http"
 
 	"github.com/hashicorp/terraform-plugin-framework/datasource"
 	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	vellumclient "terraform-provider-vellum/internal/sdk/client"
```
Comment (dvargasfuertes) [inline internal/provider/document_index/data.go:12]: I inlined just the parts of the sdk (straight copy and paste from `vellum-client-go`) that we need for this provider. We could remove the dependency on the next PR

File: internal/sdk/client/client.go
Comment (dvargas92495) [inline internal/sdk/client/client.go:1]: Everything from here on is a selective copy and paste from fern's generated go sdk

Comment (dvargasfuertes) [inline internal/sdk/client/client.go:1]: Everything from here down is a selective copy and paste from fern's generated go sdk

**Comment by dvargasfuertes** [comment]
ccing @dsinghvi

**Review by noanflaherty — APPROVED** [review]
Amazing! 🚀

**Pull request merged**